### PR TITLE
feature/agenda-settings: Implement Agenda Preferences

### PR DESCRIPTION
**Pull request by aelbenney**
In order to make the user able to customize the view of his calendar, we added a drawer in the agenda app containing these settings.

### CODE REVIEW

File: agenda-webapps/src/main/resources/locale/portlet/Agenda_en.properties [hunk 1]
```diff
@@ -114,3 +114,19 @@ agenda.details.header.menu.delete=Delete
 agenda.details.header.menu.export=Export
 
 agenda.title.addEventRecurrent=Custom recurrence
+
+agenda.settings.drawer.button.save=Save
```
Comment (boubaker) [inline agenda-webapps/src/main/resources/locale/portlet/Agenda_en.properties]: For generic labels like "Save", "cancel" .. please use a generic label in stead of duplicating it and creating a new one for each UI. You have already a "Cancel" translation here https://github.com/exoplatform/agenda/pull/26/files#diff-5627e09f74240cab54a6f4d0888af1e1R6 for example

File: agenda-webapps/src/main/webapp/vue-app/agenda/components/user-setting-agenda/UserSettingAgendaDrawer.vue [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+        </v-btn>
+      </div>
+    </template>
+  </exo-drawer>
+</template>
+
+<script>
+export default {
+  props: {
+    value: {
```
Comment (boubaker) [inline agenda-webapps/src/main/webapp/vue-app/agenda/components/user-setting-agenda/UserSettingAgendaDrawer.vue]: no need to use v-model in this drawer, use just `:settings="settings"`

File: agenda-webapps/src/main/resources/locale/portlet/Agenda_en.properties (hunk 1 above)
Comment (aelbenney) [reply agenda-webapps/src/main/resources/locale/portlet/Agenda_en.properties]: okay

File: agenda-webapps/src/main/webapp/vue-app/agenda/components/user-setting-agenda/UserSettingAgendaDrawer.vue [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+        body: JSON.stringify({
+          value: JSON.stringify(this.value),
+        }),
+      }).then(resp => {
+        if (resp && resp.ok) {
+          this.$refs.UserSettingAgendaDrawer.close();
+        }
+      })
+        .then(() => {
+          window.location.replace(window.location.href);
```
Comment (boubaker) [inline agenda-webapps/src/main/webapp/vue-app/agenda/components/user-setting-agenda/UserSettingAgendaDrawer.vue]: Why this statement ?

File: agenda-webapps/src/main/webapp/vue-app/agenda/js/CalendarService.js [hunk 4]
```diff
@@ -29,4 +29,41 @@ export function saveCalendar(calendar) {
...
+      value: value,
+    }),
+  }).then(resp => {
+    if (!resp || !resp.ok) {
+      throw new Error('Response code indicates a server error', resp);
+    }
+  });
+}
+
+export function getAgendaSetting() {
```
Comment (boubaker) [inline agenda-webapps/src/main/webapp/vue-app/agenda/js/CalendarService.js]: `getAgendaSettings` instead to say that this is about multiple settings

File: agenda-webapps/src/main/webapp/vue-app/agenda/js/CalendarService.js [hunk 5]
```diff
@@ -29,4 +29,41 @@ export function saveCalendar(calendar) {
       throw new Error('Error updating calendar');
     }
   });
+}
+
+export function saveAgendaSetting(contextKey, contextValue, scopeKey, scopeValue, key, value) {
```
Comment (boubaker) [inline agenda-webapps/src/main/webapp/vue-app/agenda/js/CalendarService.js]: `saveAgendaSettings` instead to say that this is about multiple settings

File: agenda-webapps/src/main/webapp/vue-app/agenda/components/top-toolbar/AgendaSwitchView.vue [hunk 6]
```diff
@@ -18,15 +18,15 @@
 <script>
 export default {
   props: {
-    calendarType: {
+    agendaPeriodType: {
```
Comment (boubaker) [inline agenda-webapps/src/main/webapp/vue-app/agenda/components/top-toolbar/AgendaSwitchView.vue]: if this is a data managed by the component and not passed by parent Vue component, better change this to a `data` attribute.

File: agenda-webapps/src/main/webapp/vue-app/agenda/components/user-setting-agenda/UserSettingAgendaDrawer.vue (hunk 2 above)
Comment (aelbenney) [reply agenda-webapps/src/main/webapp/vue-app/agenda/components/user-setting-agenda/UserSettingAgendaDrawer.vue]: We need the v-model to display the settings when we re-use the component in the global user settings.

File: agenda-webapps/src/main/webapp/vue-app/agenda/components/user-setting-agenda/UserSettingAgendaDrawer.vue (hunk 3 above)
Comment (aelbenney) [reply agenda-webapps/src/main/webapp/vue-app/agenda/components/user-setting-agenda/UserSettingAgendaDrawer.vue]: To refresh the agenda after applying the settings.

File: agenda-webapps/src/main/webapp/vue-app/agenda/js/CalendarService.js (hunk 4 above)
Comment (aelbenney) [reply agenda-webapps/src/main/webapp/vue-app/agenda/js/CalendarService.js]: okay

File: agenda-webapps/src/main/webapp/vue-app/agenda/js/CalendarService.js (hunk 5 above)
Comment (aelbenney) [reply agenda-webapps/src/main/webapp/vue-app/agenda/js/CalendarService.js]: oaky

File: agenda-webapps/src/main/webapp/vue-app/agenda/components/top-toolbar/AgendaSwitchView.vue (hunk 6 above)
Comment (aelbenney) [reply agenda-webapps/src/main/webapp/vue-app/agenda/components/top-toolbar/AgendaSwitchView.vue]: we need to get the value configured by the user at first.

**Review by boubaker — APPROVED** [review]

**Pull request merged**